chore(docker): update Docker image to support app development (#543)
medic/cht-conf: README.md

### Docker

CHT Conf can also be run from within a Docker container. This is useful if you already have Docker installed and do not wish to configure the various dependencies required for developing CHT apps on your local machine. The necessary dependencies are pre-packaged in the Docker image.

#### Building the image

Run the following commands to create a new project directory and build the Docker image:

```shell
mkdir cht-project
cd cht-project
curl https://raw.githubusercontent.com/medic/cht-conf/main/Dockerfile > Dockerfile
docker build -t cht-conf .
```

#### Using the image

The resulting Docker image can be used as a [VSCode Development Container](https://code.visualstudio.com/docs/devcontainers/containers) (easiest) or as a standalone Docker utility.

##### VSCode Development Container

If you want to develop CHT apps with VSCode, you can use the Docker image as a Development Container. This will allow you to use the `cht-conf` utility and its associated tech stack from within VSCode (without needing to install dependencies like NodeJS on your host system).

> [Install VSCode](https://code.visualstudio.com/) if you do not have it already.

Reviewer: ```suggestion
Install both [VSCode](https://code.visualstudio.com/) and [Windows Subsystem for Linux (WSL2)](https://learn.microsoft.com/en-us/windows/wsl/install) if you haven't already.
```

> Run the following commands from within your project directory to download the `.devcontainer.json` config file, install the [Dev Containers extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers), and open the project directory in VSCode:
> 
> ```shell
> curl https://raw.githubusercontent.com/medic/cht-conf/main/devcontainer.config/.devcontainer.json > .devcontainer.json
> code --install-extension ms-vscode-remote.remote-containers
> code -n .

Reviewer: Let's assume that if they have an existing project directory, they'll know how to use it.  But that if they don't, they'll need a lot of help, and so default to creating it for them. As well, let's quiet up `curl`. Finally let's call out they need to use the `wsl` shell not the `cmd` shell:

```suggestion
Using a WSL shell (Start -> wsl) to run the following commands. It will create a new project directory, download the `.devcontainer.json` config file, install the [Dev Containers extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers), and open the project directory in VSCode:

```shell
mkdir -p ~/cht-vscode-project && cd  ~/cht-vscode-project
curl -s https://raw.githubusercontent.com/medic/cht-conf/main/devcontainer.config/.devcontainer.json > .devcontainer.json
code --install-extension ms-vscode-remote.remote-containers
code -n .
```

Author: So, they should have already created the project directory above (when building the Docker image). Added a note here to clarify...

Also, I think it is a good idea to mention WSL for Windows, but I hesitate to make that the only language we use since anecdotally a lot of app developers use non-Windows systems too.

Reviewer: ah - dude - why am I only thinking of WSL?!  That's silly of me and you're right - so many OSes out there!  your mention of WSL is great.  I'll resolve this.

```

> 

Reviewer: I got a scary warning that feels like we should account for:

```suggestion

If this is the first time you're running VSCode in this directory, you may get a prompt:

> **Do you trust the authors of the files in this folder?**

Choose, "Yes, I trust the authors".

```

> Open the Command Palette in VSCode (_Ctrl+Shift+P_ or _Cmd+Shift+P_) and select `Reopen in Container`. This will open your workspace inside the `cht-conf` container. You can use the `cht` commands by opening a terminal in VSCode (_Ctrl+Shift+\`_ or _Cmd+Shift+\`_).

Reviewer: ```suggestion
Open the Command Palette in VSCode (_Ctrl+Shift+P_ or _Cmd+Shift+P_) and select `Reopen in Container`. This will open your workspace inside the `cht-conf` container. You can use the `cht` commands by opening a terminal in VSCode (_Ctrl+Shift+\`_ or _Cmd+Shift+\`_). If prompted "Do you trust the authors..." choose "Trust Folder & Continue".
```

Reviewer: I got a second prompt!

> Run the following command in the VSCode terminal to bootstrap your new CHT project:

Reviewer: ```suggestion
In VSCode type `ctrl` + `` ` ``  to open a terminal in your Dev Container, and run the following command to bootstrap your new CHT project:
```

Author: This feels unnecessary since we just told them how to open the terminal on the previous line...  :thinking:

Reviewer: oh - i didn't rtfm - sorry!  Agreed, redundant, I'll self resolve this.

```shell
mkdir -p ~/cht-project
cd ~/cht-project
curl https://raw.githubusercontent.com/medic/cht-conf/main/Dockerfile > Dockerfile
docker build -t cht-conf .
```

#### Using the image

The resulting Docker image can be used as a [VSCode Development Container](https://code.visualstudio.com/docs/devcontainers/containers) (easiest) or as a standalone Docker utility.

##### VSCode Development Container

If you want to develop CHT apps with VSCode, you can use the Docker image as a Development Container. This will allow you to use the `cht-conf` utility and its associated tech stack from within VSCode (without needing to install dependencies like NodeJS on your host system).

[Install VSCode](https://code.visualstudio.com/) if you do not have it already.

Using the terminal (or the WLS shell on Windows: _Start > wsl_), run the following commands from within your project directory (created above) to download the `.devcontainer.json` config file, install the [Dev Containers extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers), and open the project directory in VSCode:

> 
> ```shell
> curl https://raw.githubusercontent.com/medic/cht-conf/main/devcontainer.config/.devcontainer.json > .devcontainer.json
> code --install-extension ms-vscode-remote.remote-containers
> code -n .
> ```

Reviewer: Minor point to keep the whole "copy paste and it just works!" vibe goin by adding a `cd` and then still adding a `-s` to quiet up `curl`:

```suggestion

```shell
cd ~/cht-project
curl -s https://raw.githubusercontent.com/medic/cht-conf/main/devcontainer.config/.devcontainer.json > .devcontainer.json
code --install-extension ms-vscode-remote.remote-containers
code -n .
```
```

> 

Reviewer: oops! I think my suggestion last time ate the ` ``` ` needed on line 67.  I can't figure out how to make a suggestion  that has it in there, but it looks link this now:

![image](https://user-images.githubusercontent.com/8253488/223195581-597e7b40-cc2d-4f7f-a462-143d3ac7069a.png)

Author: Oh good catch!  Pushed a fix.
